modules/csvparser/filterx-func-parse-csv.c

```diff
@@ -0,0 +1,346 @@
+/*
+ * Copyright (c) 2024 shifter
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 as published
+ * by the Free Software Foundation, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * As an additional exemption you are allowed to compile & link against the
+ * OpenSSL libraries as published by the OpenSSL project. See the file
+ * COPYING for details.
+ *
+ */
+
+#include "filterx-func-parse-csv.h"
+#include "filterx/object-string.h"
+#include "filterx/object-primitive.h"
+#include "filterx/expr-literal.h"
+#include "filterx/filterx-eval.h"
+#include "filterx/filterx-globals.h"
+#include "filterx/object-json.h"
+#include "filterx/object-message-value.h"
+#include "filterx/object-null.h"
+#include "filterx/filterx-object.h"
+#include "filterx/object-dict-interface.h"
+#include "filterx/object-list-interface.h"
+
+#include "scanner/csv-scanner/csv-scanner.h"
+#include "parser/parser-expr.h"
+#include "scratch-buffers.h"
+#include "str-utils.h"
+#include "csvparser.h"
+
+typedef struct FilterXFunctionParseCSV_
+{
+  FilterXFunction super;
+  FilterXExpr *msg;
+  CSVScannerOptions options;
+  FilterXExpr *columns;
+} FilterXFunctionParseCSV;
+
+static gboolean
+_parse_columns(FilterXFunctionParseCSV *self, GList **col_names)
+{
+  gboolean result = FALSE;
+  if (!self->columns)
+    return TRUE;
+  FilterXObject *cols_obj = filterx_expr_eval(self->columns);
+  if (!cols_obj)
+    return FALSE;
+
+  if (filterx_object_is_type(cols_obj, &FILTERX_TYPE_NAME(null)))
+    {
+      result = TRUE;
+      goto exit;
+    }
```

Comment:
I don't think we should support these now that `columns` is an optional argument. I can't imagine someone setting `columns=null`.